probe-rs/src/architecture/arm/memory/adi_v5_memory_interface.rs

```diff
@@ -11,29 +11,53 @@ use crate::{CommunicationInterface, Error};
 use std::convert::TryInto;
 use std::ops::Range;
 
-pub trait ArmProbe: SwdSequence {
-    fn read_8(&mut self, ap: MemoryAp, address: u64, data: &mut [u8]) -> Result<(), Error>;
+pub trait ArmMemoryAccess: SwdSequence {
```

Comment:
Well, The question is how naming conventions are. Because It's two different traits which are named correctly independently but one requires the other and allows access to the others functions making naming seem a bit off.
Not sure how conventions should be here.